cmd/docker-app/inspect.go

```diff
@@ -15,21 +16,47 @@ var (
 	inspectEnv            []string
 )
 
+// inspectCmd represents the inspect command
 func inspectCmd(dockerCli command.Cli) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "inspect [<app-name>] [-s key=value...] [-f parameters-file...]",
 		Short: "Shows metadata, parameters and a summary of the compose file for a given application",
 		Args:  cli.RequiresMaxArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			app, err := packager.Extract(firstOrEmpty(args),
-				types.WithParametersFiles(inspectParametersFile...),
-			)
+			muteDockerCli(dockerCli)
+			appname := firstOrEmpty(args)
+			bundle, err := resolveBundle(dockerCli, "", appname)
 			if err != nil {
 				return err
 			}
-			defer app.Cleanup()
-			argParameters := cliopts.ConvertKVStringsToMap(inspectEnv)
-			return inspect.Inspect(dockerCli.Out(), app, argParameters)
+			params, err := parameters.LoadFiles(inspectParametersFile)
+			if err != nil {
+				return err
+			}
+			overrides, err := parameters.FromFlatten(cliopts.ConvertKVStringsToMap(inspectEnv))
+			if err != nil {
+				return err
+			}
+			if params, err = parameters.Merge(params, overrides); err != nil {
+				return err
+			}
+			c, err := claim.New("inspect")
+			if err != nil {
+				return err
+			}
+			driverImpl, err := prepareDriver(dockerCli)
+			if err != nil {
+				return err
+			}
+			c.Bundle = bundle
+			c.Parameters = stringsKVToStringInterface(params.Flatten())
+
+			a := &action.RunCustom{
+				Action: "inspect",
+				Driver: driverImpl,
+			}
+			err = a.Run(c, map[string]string{"docker.context": ""}, dockerCli.Out())
```

Comment:
We might need to do that for every credential declared (the empty credential input)

Reply:
You mean declared in the bundle?

Reply:
But as `inspect` is a `docker-app` CNAB custom action, we don't expect it to have more than the `docker.context` credential, right? And I guess we don't plan to support `inspect` on other CNABs, even if they declare this particular custom action.

Reply:
If any CNAB author find this trick interesting and want to use this behavior as a "de-facto" standard extension (or if my PR about stateless CNAB action pass, or also when we will have a CNAB composition tool), we'll need to do that.

Reply:
Sounds like custom actions might need to be namespaced, otherwise you can never confidently call any of them unless you have other context telling you it is safe. 

Is that covered by the CNAB spec? It might mean a spec update to recommend e.g. `com.docker.app.inspect` or some such (we could unilaterally do it that way anyway I guess).

Reply:
cc @garethr WDYT?

Reply:
I like the idea if namespacing actions.

As Simon notes, with stateless actions I think their is room for patterns to emerge. But being cautious about the root namespace is probably prudent at the moment.

So +1 from me I think.